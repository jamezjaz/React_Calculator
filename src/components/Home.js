import React from 'react';
import '../styles/Home.css';
import logo from '../assets/logo.png';

const Home = () => (
  <div className="home">
    <h2 className="heading">Hello there!</h2>
    <div className="welcome">
      <h4>
        You&apos;re welcome to our page
      </h4>
      <img src={logo} alt="logo" className="logo" />
    </div>
    <p>
      In 1642, the first true “calculator” was invented: one that performed calculations
      through a clockwork-type of mechanism. The Pascal calculator, invented by French inventor
      and mathematician Blaise Pascal, was lauded for attempting arithmetic calculations
      previously thought impossible. But unfortunately, they were difficult to produce and very
      few were ever made. The mechanical calculator then invented by Thomas de Colmar in the
      mid-nineteenth century, and subsequent others, were easier to produce, but extremely large
      and bulky–not at all the pocket calculators we know today.
    </p>
    <p>
      Curt Herzstark invented the first handheld, mechanic calculator in 1945, from a design he had
      created in 1938. Shaped like a stout pepper grinder, the Curta calculators were produced in
      large quantities until 1970, when a company in Japan invented the first digital pocket
      calculator and the demand for the antiquated device faded. Companies in the United States,
      such as Texas Instruments, adapted the design of the Japanese device, and enhanced it by
      creatingthe graphing calculators we know today.
    </p>
  </div>
);

export default Home;
