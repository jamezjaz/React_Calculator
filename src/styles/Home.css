.home {
  font-size: 20px;
  padding: 20px;
  line-height: 1.5;
}

.welcome {
  display: flex;
  justify-content: space-between;
}

.logo {
  height: 20vmin;
  pointer-events: none;
  margin-right: 100px;
}

@media (prefers-reduced-motion: no-preference) {
  .logo {
    animation: logo-spin infinite 20s linear;
  }
}

/* .App-header {
  background-color: #282C34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61DAFB;
} */

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
