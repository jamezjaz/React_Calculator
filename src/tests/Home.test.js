import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });
import Home from '../components/Home';

describe('<Home />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('it contains a h2 element', () => {
    let wrapper = shallow(<Home />);
    const h2 = <h2>Hello there!</h2>;
    expect(wrapper.containsMatchingElement(h2)).toBe(true);
  });

  it('h2 element has a text', () => {
    let wrapper = shallow(<Home />);
    const h2 = wrapper.find('.heading');
    expect(h2.text()).toEqual('Hello there!');
  });

  it('should fail if the text on the h2 element is incorrect', () => {
    let wrapper = shallow(<Home />);
    const h2Text = <h2>Hey there!</h2>;
    expect(wrapper.containsMatchingElement(h2Text)).toBe(false);
  });
});