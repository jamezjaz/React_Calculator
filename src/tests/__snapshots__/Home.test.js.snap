// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<div
  className="home"
>
  <h2
    className="heading"
  >
    Hello there!
  </h2>
  <div
    className="welcome"
  >
    <h4>
      You're welcome to our page
    </h4>
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
  </div>
  <p>
    In 1642, the first true “calculator” was invented: one that performed calculations through a clockwork-type of mechanism. The Pascal calculator, invented by French inventor and mathematician Blaise Pascal, was lauded for attempting arithmetic calculations previously thought impossible. But unfortunately, they were difficult to produce and very few were ever made. The mechanical calculator then invented by Thomas de Colmar in the mid-nineteenth century, and subsequent others, were easier to produce, but extremely large and bulky–not at all the pocket calculators we know today.
  </p>
  <p>
    Curt Herzstark invented the first handheld, mechanic calculator in 1945, from a design he had created in 1938. Shaped like a stout pepper grinder, the Curta calculators were produced in large quantities until 1970, when a company in Japan invented the first digital pocket calculator and the demand for the antiquated device faded. Companies in the United States, such as Texas Instruments, adapted the design of the Japanese device, and enhanced it by creatingthe graphing calculators we know today.
  </p>
</div>
`;
